### ISSUE TITLE
Login using fresh account with pending invites fails

### ISSUE THREAD

**Issue by ptman** [vector-im/riot-web#8692]
### Description

Riot Web 1.0 log in using fresh account fails.

### Steps to reproduce

- I upgraded riot-web from 0.17.9 to 1.0.0
- Also updated synapse to 0.99.1.1
- Old sessions keep working
- Old users can sign in
- User freshly created with register_new_matrix_user fails to log in

Log: not sent, don't get that far

Browser console (a bit sanitized for private info):

```
Now tracking device list for @user:domain.tld
rageshake.js:99 Starting key download for 
["@user:domain.tld"]
rageshake.js:99 MatrixClient sync state => SYNCING
rageshake.js:99 Presence: online
rageshake.js:99 Saving device tracking data at token s13326_215446_22_12261_11907_62_1_667_84
Unread.js:56 Uncaught TypeError: e.getEventReadUpTo is not a function
    at Object.doesRoomHaveUnreadMessages (Unread.js:56)
    at Object.makeRoomTile (RoomSubList.js:188)
    at Array.map (<anonymous>)
    at t.value (LazyRenderList.js:120)
    at l._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:796)
    at l._renderValidatedComponent (ReactCompositeComponent.js:819)
    at l.performInitialMount (ReactCompositeComponent.js:359)
    at l.mountComponent (ReactCompositeComponent.js:255)
    at Object.mountComponent (ReactReconciler.js:43)
    at X.mountChildren (ReactMultiChild.js:234)
    at X._createInitialChildren (ReactDOMComponent.js:701)
    at X.mountComponent (ReactDOMComponent.js:520)
    at Object.mountComponent (ReactReconciler.js:43)
    at X.mountChildren (ReactMultiChild.js:234)
    at X._createInitialChildren (ReactDOMComponent.js:701)
    at X.mountComponent (ReactDOMComponent.js:520)
ReactDOMComponentTree.js:111 Uncaught TypeError: Cannot read property '__reactInternalInstance$acy84h28mye' of null
    at Object.p [as getClosestInstanceFromNode] (ReactDOMComponentTree.js:111)
    at f (ReactEventListener.js:36)
    at p (ReactEventListener.js:65)
    at u.perform (Transaction.js:141)
    at Object.batchedUpdates (ReactDefaultBatchingStrategy.js:60)
    at Object.batchedUpdates (ReactUpdates.js:95)
    at dispatchEvent (ReactEventListener.js:145)
indexeddb-local-backend.js:488 Persisting sync data up to  s13326_215446_22_12261_11907_62_1_667_84
rageshake.js:99 Looking for queued outgoing room key requests
ReactDOMComponentTree.js:111 Uncaught TypeError: Cannot read property '__reactInternalInstance$acy84h28mye' of null
    at Object.p [as getClosestInstanceFromNode] (ReactDOMComponentTree.js:111)
    at f (ReactEventListener.js:36)
    at p (ReactEventListener.js:65)
    at u.perform (Transaction.js:141)
    at Object.batchedUpdates (ReactDefaultBatchingStrategy.js:60)
    at Object.batchedUpdates (ReactUpdates.js:95)
    at dispatchEvent (ReactEventListener.js:145)
rageshake.js:99 No more outgoing room key requests
ReactDOMComponentTree.js:111 Uncaught TypeError: Cannot read property '__reactInternalInstance$acy84h28mye' of null
    at Object.p [as getClosestInstanceFromNode] (ReactDOMComponentTree.js:111)
    at f (ReactEventListener.js:36)
    at p (ReactEventListener.js:65)
    at u.perform (Transaction.js:141)
    at Object.batchedUpdates (ReactDefaultBatchingStrategy.js:60)
    at Object.batchedUpdates (ReactUpdates.js:95)
    at dispatchEvent (ReactEventListener.js:145)
ReactCompositeComponent.js:742 Uncaught TypeError: Cannot read property '_currentElement' of null
    at l._updateRenderedComponent (ReactCompositeComponent.js:742)
    at l._performComponentUpdate (ReactCompositeComponent.js:721)
    at l.updateComponent (ReactCompositeComponent.js:642)
    at l.receiveComponent (ReactCompositeComponent.js:544)
    at Object.receiveComponent (ReactReconciler.js:122)
    at Object.updateChildren (ReactChildReconciler.js:107)
    at X._reconcilerUpdateChildren (ReactMultiChild.js:209)
    at X._updateChildren (ReactMultiChild.js:308)
    at X.updateChildren (ReactMultiChild.js:295)
    at X._updateDOMChildren (ReactDOMComponent.js:944)
    at X.updateComponent (ReactDOMComponent.js:758)
    at X.receiveComponent (ReactDOMComponent.js:720)
    at Object.receiveComponent (ReactReconciler.js:122)
    at Object.updateChildren (ReactChildReconciler.js:107)
    at X._reconcilerUpdateChildren (ReactMultiChild.js:209)
    at X._updateChildren (ReactMultiChild.js:308)
rageshake.js:99 got keys for @user:domain.tld
{YXVSXCBBNW: {…}, DWATWHPULM: {…}, OSBZOLQZFZ: {…}, PPDWEWLHZA: {…}, YLXHSQZALQ: {…}, …}
rageshake.js:99 Completed key download for @user:domain.tld
rageshake.js:99 Device list for @user:domain.tld now up to date
rageshake.js:99 Saving device tracking data at token s13326_215446_22_12261_11907_62_1_667_84
```

### Version information

- **Platform**: web
- **Browser**: Firefox, Chrome
- **OS**: Linux
- **URL**: Private instance, riot 1.0.0, config.json sets default home server

**Comment by ptman** [comment]
So this may have had something to do with invites that I made to the account before first login.

I logged in using riot 0.17.9 and accepted the invites. After that login succeeded.

**Comment by bwindels** [comment]
Thanks for the report!
Did you have any community invites by any chance?

**Comment by ptman** [comment]
yes, there was also a community invite

**Comment by bwindels** [comment]
Caused by https://github.com/matrix-org/matrix-react-sdk/pull/2629, making a fix now.

**Comment by bwindels** [comment]
just released 1.0.1 that should fix this. Please reopen if you still see this.
